unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1151,34 +1134,64 @@ public class OptionalVector3 {
             }
 
             // let's look in the agent's set of actions for the action
-            this.activeAgent().ProcessControlCommand(controlCommand: controlCommand);
-        }
+			this.activeAgent().ProcessControlCommand(controlCommand: controlCommand);
+		}
     }
 
-    // Extra helper functions
-    protected string LoadStringVariable(string variable, string name) {
-        string envVarName = ENVIRONMENT_PREFIX + name.ToUpper();
-        string envVarValue = Environment.GetEnvironmentVariable(envVarName);
-        return envVarValue == null ? variable : envVarValue;
-    }
+    public BaseFPSAgentController GetActiveAgent() {
+		return this.agents[activeAgentId];
+	}
 
-    protected int LoadIntVariable(int variable, string name) {
-        string envVarName = ENVIRONMENT_PREFIX + name.ToUpper();
-        string envVarValue = Environment.GetEnvironmentVariable(envVarName);
-        return envVarValue == null ? variable : int.Parse(envVarValue);
-    }
+	private int parseContentLength(string header) {
+		// Debug.Log("got header: " + header);
+		string[] fields = header.Split(new char[]{'\r','\n'});
+		foreach(string field in fields) {
+			string[] elements = field.Split(new char[]{':'});
+			if (elements[0].ToLower() == "content-length") {
+				return Int32.Parse(elements[1].Trim());
+			}
+		}
 
-    protected float LoadFloatVariable(float variable, string name) {
-        string envVarName = ENVIRONMENT_PREFIX + name.ToUpper();
-        string envVarValue = Environment.GetEnvironmentVariable(envVarName);
-        return envVarValue == null ? variable : float.Parse(envVarValue);
-    }
+		return 0;
+	}
 
-    protected bool LoadBoolVariable(bool variable, string name) {
-        string envVarName = ENVIRONMENT_PREFIX + name.ToUpper();
-        string envVarValue = Environment.GetEnvironmentVariable(envVarName);
-        return envVarValue == null ? variable : bool.Parse(envVarValue);
-    }
+	private BaseFPSAgentController activeAgent() {
+		return this.agents[activeAgentId];
+	}
+
+	private void ProcessControlCommand(string msg) {
+        DynamicServerAction controlCommand = new DynamicServerAction(jsonMessage: msg);
+        this.ProcessControlCommand(controlCommand);
+	}
+
+	// Extra helper functions
+	protected string LoadStringVariable(string variable, string name)
+	{
+		string envVarName = ENVIRONMENT_PREFIX + name.ToUpper();
+		string envVarValue = Environment.GetEnvironmentVariable(envVarName);
+		return envVarValue == null ? variable : envVarValue;
+	}
+
+	protected int LoadIntVariable(int variable, string name)
+	{
```

Comment:
Can you check your formatter settings? The first brace shouldn't be on a new line if .editorconfig exists in the root.

Reply:
Fixed